GetComponents

```diff
@@ -1493,7 +1493,15 @@ sub handle_git {
 
         # clone the git repo
         elsif ( !-e $repo_loc ) {
-            $cmd = "$git clone$shallow $url '$repo_loc'";
+            my $branch_opts = "";
+            if ( $#branches == 0 ) {
+                $branch_opts = " --branch '$branches[0]'";
+            }
+            die "Error: Multiple branch checkout is no longer supported" if($#branches > 0);
+            if ( $SHALLOW_CLONE and $#branches <= 0) {
```

Comment:
I'd probably do `== 0` rather than `<= 0` since you already checked that it cannot be `> 0` above. In fact I would probably prefer `scalar @branches` since that one gives the actual number (so 0 for empty, 1 for 1 elements) instead of the highest valid index the way way `$#branches` does. What do other places in GetComponents do?